frontend/www/js/omegaup/arena/ranking.ts

```diff
@@ -276,3 +277,161 @@ export function onRankingChanged({
     lastTimeUpdated: scoreboard.time,
   };
 }
+
+export function mergeRankings({
+  scoreboard,
+  originalScoreboardEvents,
+  navbarProblems,
+}: {
+  scoreboard: types.Scoreboard;
+  originalScoreboardEvents: types.ScoreboardEvent[];
+  navbarProblems: types.NavbarProblemsetProblem[];
+}): {
+  mergedScoreboard: types.Scoreboard;
+  originalContestEvents: types.ScoreboardEvent[];
+} {
+  // This clones virtualContestData to data so that virtualContestData values
+  // won't be overriden by processes below
+  const data = JSON.parse(JSON.stringify(scoreboard)) as types.Scoreboard;
+  const dataRanking: (types.ScoreboardRankingEntry & {
+    virtual?: boolean;
+  })[] = data.ranking;
+  const events = originalScoreboardEvents ?? [];
```

Comment:
```suggestion
  const events: types.ScoreboardEvent = originalScoreboardEvents ?? [];
```

(porque tiene un arreglo literal sin tipo en el lado derecho).

ó

```suggestion
  const events = originalScoreboardEvents;
```

porque bajo ninguna circunstancia el `??` se va a usar: el tipo no está declarado como nulable, y [`??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) sólo funciona con `null` y `undefined`.

Reply:
Mejor lo quito.